src/hotspot/os/posix/os_posix.cpp

```diff
@@ -349,13 +329,39 @@ char* os::reserve_memory_aligned(size_t size, size_t alignment, int file_desc) {
       os::release_memory(extra_base + begin_offset + size, end_offset);
   }
 
-  if (file_desc != -1) {
-    // After we have an aligned address, we can replace anonymous mapping with file mapping
-    if (replace_existing_mapping_with_file_mapping(aligned_base, size, file_desc) == NULL) {
-      vm_exit_during_initialization(err_msg("Error in mapping Java heap at the given filesystem directory"));
-    }
-    MemTracker::record_virtual_memory_commit((address)aligned_base, size, CALLER_PC);
+  return aligned_base;
+}
+
+// Multiple threads can race in this code, and can remap over each other with MAP_FIXED,
```

Comment:
I would like to persist the comment (as the race is not evident for the reader). As for the placement, the comment describes an alternative implementation of this function and provides rationale why the current one is taken. chop_extra_memory is a step of the implementation, that has no alternative. I think it would be wrong to move the comment there. But I've added a comment for chop_extra_memory for clarity.

Reply:
Okay.